processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -167,12 +167,17 @@ public boolean isLimited()
       }
     }
 
+    if (!sortingNeeded) {
+      // If granularity is ALL, sortByDimsFirst doesn't change the sorting order.
+      sortingNeeded = granularity.equals(Granularities.ALL) || sortByDimsFirst;
```

Comment:
so if , granularity = ALL and sortByDimsFirst = false, then "sortingNeeded" gets set to true while in older versions it would've been false. I see the comment but still not sure why this change is necessary.

Reply:
Sorry, this is a bug. Fixed it and added a test.